### ISSUE TITLE
Unable to set column and row heading rows in MS Word tables

### ISSUE THREAD

**Issue by andrew-l-d** [nvaccess/nvda#15501]
### Steps to reproduce:
1. Create a Word document
2. Create a table of - say - three columns and three rows
3. Add data to the table such as Last name, First name and phone to cells in top row and relevant info below
4. With the cursor in the top left cell, press NVDA+Shift+c to set top row as column headers
### Actual behavior:
Last name
alt+ctrl+right arrow
First name
alt+ctrl+left arrow
column 1  Last name
shift+NVDA+c
right arrow
a
Alfa
left arrow
L
Lima
left arrow
C
Charlie
### Expected behavior:
Top row is set as column headers.  And it worked correctly very recently.  I wonder if something has changed in Word or, just as likely, I have missed something.  Changing items in the Table Design tab of the Ribbon works. 
#### NVDA installed/portable/running from source:
Installed or portable
#### NVDA version:
2022.4 to 2023.3 beta 2
#### Windows version:
10
#### Name and version of other software in use when reproducing the issue:
MS Word in Office 365, both 32 and 64 bit versions
#### Other information about your system:
Up to date on both machines.
#### Does the issue still occur after restarting your computer?
Yes
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
See above
#### If NVDA add-ons are disabled, is your problem still occurring?
Yes
##. ## Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
Yes

**Comment by zstanecic** [comment]
Hi,

It is not a bug. Microsoft office 365 sets headers automatically. The option you mentioned works for older NVDA.

**Comment by Adriani90** [comment]
Is table headers reporting enabled n nvda document formating settings?
And can you reproduce this issue with and without using UIA?

**Comment by andrew-l-d** [comment]
I knew there was something I wanted to check before raising the ticket.  Yes, reporting of table headers is on - wouldn't fall for that one.  But with UI Automation set to always it does not work.  When set to Only Where Necessary or Where Suitable it does.  Perhaps Section 11.1.1 of the Manual should mention that.  I will close the ticket once the self-admonishment is complete.

Andrew

**Comment by cary-rowen** [comment]
@zstanecic 
> It is not a bug. Microsoft office 365 sets headers automatically. The option you mentioned works for older NVDA.
Thanks, can you tell me more specifically? I also think this is a bug. How to make word automatically generate row and column titles.

**Comment by andrew-l-d** [comment]
Firstly, it is not a bug as such,but the feature does not work if Use UI Automation to access Microsoft Word document controls in NVDA's Advanced settings is set to always.  That is something I intended to check before raising the ticket and forgot.
Now to Word settings.  I would not call it automatic.  If you go into table properties and check Repeat Header Row at top of each Page (even if the table fits on one page), NVDA will treat the top row as headings.  In Word 365 (not sure about the non-subscription version), there is another option.  When the cursor is in the table an extra two tabs appear in the Ribbon.  In the Table Design tab there is a series of checkboxes.  Checking Show Special Formatting for First Row will cause NVDA to treat it as a header row.  Similarly, checking Show Special Formatting for First Column will have it treated as row headings.  Lately I have found both checked by default, which is often not appropriate.  Checking both of those checkboxes is especially important if converting to PDF, as the cells in the PDF table will be tagged correctly.

Andrew
